src/content/docs/en/guides/images.mdx

```diff
@@ -4,159 +4,677 @@ description: Learn how to use images in Astro.
+If you require more control over your image attributes, we recommend using the `.mdx` file format, which allows you to include Astro's `<Image />` component or a JSX `<img />` tag in addition to the Markdown syntax. Use the [MDX integration](/en/guides/integrations-guide/mdx/) to add support for MDX to Astro.
 
-Additionally, you can import and use images located in your project's `src/` directory, as you would in Astro components.
+## Images in MDX files
 
-```mdx title="src/pages/post-1.mdx"
+You can use Astro's `<Image />` component and JSX `<img />` tags in your `.mdx` files by importing both the component and your image. Use them just as they are [used in `.astro` files](#images-in-astro-files).
 
-import rocket from '../images/rocket.svg';
+Additionally, there is support for [standard Markdown `![alt](src)` syntax](#images-in-markdown-files) with no import required.
+
+```mdx title="src/pages/post-1.mdx"
+---
+title: My Page title
+---
+import { Image } from 'astro:assets';
+import rocket from '../assets/rocket.png';
 
 # My MDX Page
 
-// Local image stored at src/images/rocket.svg
-<img src={rocket} alt="A rocketship in space."/>
+// Local image stored in src/assets/
+<Image src={rocket} alt="A rocketship in space."/>
+<img src={rocket.src} alt="A rocketship in space." />
+![A rocketship in space](../assets/rocket.png)
+
+// Image stored in public/images/
+<Image src="/images/stars.png" alt="A starry night sky." />
+<img src="/images/stars.png" alt="A starry night sky." />
+![A starry night sky.](/images/stars.png)
 
-// Local image stored at public/assets/stars.png
-![A starry night sky.](/assets/stars.png)
-<img src="/assets/stars.png" alt="A starry night sky." />
 
 // Remote image on another server
-![Astro](https://docs.astro.build/assets/logomark-light.png)
-<img src="https://docs.astro.build/assets/logomark-light.png" width="25" alt="Astro" />
+<Image src="https://example.com/images/remote-image.png" />
+<img src="https://example.com/images/remote-image.png" />
+![Astro](https://example.com/images/remote-image.png)
+
 ```
 
-### In UI Framework Components
+## Images in Content Collections
 
-When adding images in a [UI framework component](/en/core-concepts/framework-components/) (e.g React, Svelte), use the image syntax appropriate for that particular component's framework.
+You can declare an associated image for a content collections entry, such as a blog post's cover image, in your frontmatter using its path relative to the current folder:
 
-## Where to store images
+```md title="src/content/blog/my-post.md" {3}
+---
+title: "My first blog post"
+cover: "./firstpostcover.jpeg" # will resolve to "src/content/blog/firstblogcover.jpeg"
+coverAlt: "A photograph of a sunset behind a mountain range."
+---
+
+This is a blog post
+```
 
-### `src/`
+The `image` helper for content collections lets you validate the image metadata using Zod.
```

Comment:
This schema is actually required for the image to be resolved, otherwise it's just treated as a normal string. It being able to validate the image dimensions is a bonus